docs/user/managing-fleets.md

```diff
@@ -0,0 +1,117 @@
+
+### Selecting Devices into a Fleet on the CLI
+
+Before adding a label selector to a fleet, you can test it by getting the list of devices matching that selector. To test the selector `type=pos-terminal, stage=development` in the above example, you would run:
+
+```console
+flightctl get devices -l type=pos-terminal -l stage=development
+```
+
+If the list of returned devices is OK, you could then define a `development-pos-terminals` fleet selecting these like this:
+
+```yaml
+apiVersion: v1alpha1
+kind: Fleet
+metadata:
+  name: development-pos-terminals
+spec:
+  selector:
+    matchLabels:
+      type: pos-terminal
+      stage: development
+[...]
+```
+
+After applying the change, you can check whether there's an overlap with another fleet's selector like this (assuming you have installed the `jq` command):
+
+```console
+flightctl get fleets/development-pos-terminals -o json | jq -r '.status.conditions[] | select(.type=="OverlappingSelectors").status'
+```
+
+You should see the output:
+
+```console
+False
+```
+
+## Defining Device Templates
+
+A fleet's device template contains a device specification that gets applied to all devices in the fleet when the template gets updated. In other words, you could take an existing device's specification and create a new fleet whose template is a copy of that specification. You can then join that device to the fleet and join additional devices to the fleet and Flight Control would enforce that they all eventually have the exact same specification.
+
+For example, you could specify in a fleet's device template that all devices in the fleet shall run the OS image `quay.io/flightctl/rhel:9.5`. The Flight Control service would then roll out this specification to all devices in the fleet and the Flight Control agents would update the devices accordingly. The same would apply to the other specification items described in [Managing Devices](managing-devices.md).
+
+However, it would be impractical if *all* of a fleet's devices had to have the *exact same specification*. Flight Control therefore allows templates to contain placeholders that get filled in based on a device's name or label values.
+
+Here are some examples what you can do with placeholders in device templates:
+
+* You can label devices by their deployment stage (say, `stage: testing` and `stage: production`) and then use the label with the key `stage` as placeholder when referencing the OS image to use (say, `quay.io/myorg/myimage:latest-{{ .Labels["stage"] }}`) or when referencing a folder with configuration in a Git repository.
+* You can label devices by deployment site (say, `site: factory-berlin` and `site: factory-madrid`) and then use the label with the key `site` as parameter when referencing the secret with network access credentials in Kubernetes.
+
+The following placeholders are supported:
+
+| Placeholder | Gets replaced by |
+| ----------- | ---------------- |
+| `{{ .Name }}` | The name of a device. |
```

Comment:
This syntax does not seem to match the current implementation.

See the examples in https://github.com/flightctl/flightctl/blob/main/internal/tasks/config_helpers_test.go#L75

Reply:
Good catch, thank you!